From fc683ee914f0a6183ac0742b92f5eb0019f24be8 Mon Sep 17 00:00:00 2001
From: Raymond Ngun <raymond.ngun@cypress.com>
Date: Mon, 2 Nov 2020 09:53:06 -0800
Subject: [PATCH] Add ARCH_TEST_RSA_2048_GEN_KEY

PSoC64 needs a way to disable RSA2048 key generation since it
is an unbounded operation and may take an extremely long time
to complete.

Signed-off-by: Raymond Ngun <raymond.ngun@infineon.com>
---
 api-tests/dev_apis/crypto/test_c016/test_data.h                 | 2 ++
 .../platform/targets/tgt_dev_apis_stdc/nspe/pal_crypto_config.h | 1 +
 .../targets/tgt_dev_apis_tfm_an521/nspe/pal_crypto_config.h     | 1 +
 .../targets/tgt_dev_apis_tfm_an524/nspe/pal_crypto_config.h     | 1 +
 .../targets/tgt_dev_apis_tfm_an539/nspe/pal_crypto_config.h     | 1 +
 .../targets/tgt_dev_apis_tfm_musca_a/nspe/pal_crypto_config.h   | 1 +
 .../targets/tgt_dev_apis_tfm_musca_b1/nspe/pal_crypto_config.h  | 1 +
 .../targets/tgt_dev_apis_tfm_musca_s1/nspe/pal_crypto_config.h  | 1 +
 .../targets/tgt_dev_apis_tfm_nrf/nspe/pal_crypto_config.h       | 1 +
 .../targets/tgt_dev_apis_tfm_psoc64/nspe/pal_crypto_config.h    | 1 +
 10 files changed, 11 insertions(+)

diff --git a/api-tests/dev_apis/crypto/test_c016/test_data.h b/api-tests/dev_apis/crypto/test_c016/test_data.h
index 287a8dae0a..f080dc85ba 100644
--- a/api-tests/dev_apis/crypto/test_c016/test_data.h
+++ b/api-tests/dev_apis/crypto/test_c016/test_data.h
@@ -108,6 +108,7 @@ static const test_data check1[] = {
 
 #ifdef ARCH_TEST_RSA
 #ifdef ARCH_TEST_RSA_2048
+#ifdef ARCH_TEST_RSA_2048_GEN_KEY
 #ifdef ARCH_TEST_RSA_PKCS1V15_SIGN_RAW
 {
     .test_desc       = "Test psa_generate_key with RSA 2048 Keypair\n",
@@ -121,6 +122,7 @@ static const test_data check1[] = {
 #endif
 #endif
 #endif
+#endif
 
 #ifdef ARCH_TEST_ECC_CURVE_SECP224R1
 #ifdef ARCH_TEST_ASYMMETRIC_ENCRYPTION
diff --git a/api-tests/platform/targets/tgt_dev_apis_stdc/nspe/pal_crypto_config.h b/api-tests/platform/targets/tgt_dev_apis_stdc/nspe/pal_crypto_config.h
index 5b7d7851aa..de6c325f25 100644
--- a/api-tests/platform/targets/tgt_dev_apis_stdc/nspe/pal_crypto_config.h
+++ b/api-tests/platform/targets/tgt_dev_apis_stdc/nspe/pal_crypto_config.h
@@ -37,6 +37,7 @@
 #define ARCH_TEST_RSA
 #define ARCH_TEST_RSA_1024
 #define ARCH_TEST_RSA_2048
+#define ARCH_TEST_RSA_2048_GEN_KEY
 #define ARCH_TEST_RSA_3072
 
 /**
diff --git a/api-tests/platform/targets/tgt_dev_apis_tfm_an521/nspe/pal_crypto_config.h b/api-tests/platform/targets/tgt_dev_apis_tfm_an521/nspe/pal_crypto_config.h
index c936bdd273..c04d2cf0fa 100644
--- a/api-tests/platform/targets/tgt_dev_apis_tfm_an521/nspe/pal_crypto_config.h
+++ b/api-tests/platform/targets/tgt_dev_apis_tfm_an521/nspe/pal_crypto_config.h
@@ -39,6 +39,7 @@
 #define ARCH_TEST_RSA
 #define ARCH_TEST_RSA_1024
 #define ARCH_TEST_RSA_2048
+#define ARCH_TEST_RSA_2048_GEN_KEY
 #define ARCH_TEST_RSA_3072
 #endif
 #endif
diff --git a/api-tests/platform/targets/tgt_dev_apis_tfm_an524/nspe/pal_crypto_config.h b/api-tests/platform/targets/tgt_dev_apis_tfm_an524/nspe/pal_crypto_config.h
index c936bdd273..c04d2cf0fa 100644
--- a/api-tests/platform/targets/tgt_dev_apis_tfm_an524/nspe/pal_crypto_config.h
+++ b/api-tests/platform/targets/tgt_dev_apis_tfm_an524/nspe/pal_crypto_config.h
@@ -39,6 +39,7 @@
 #define ARCH_TEST_RSA
 #define ARCH_TEST_RSA_1024
 #define ARCH_TEST_RSA_2048
+#define ARCH_TEST_RSA_2048_GEN_KEY
 #define ARCH_TEST_RSA_3072
 #endif
 #endif
diff --git a/api-tests/platform/targets/tgt_dev_apis_tfm_an539/nspe/pal_crypto_config.h b/api-tests/platform/targets/tgt_dev_apis_tfm_an539/nspe/pal_crypto_config.h
index 132240a68d..8a902ce0aa 100644
--- a/api-tests/platform/targets/tgt_dev_apis_tfm_an539/nspe/pal_crypto_config.h
+++ b/api-tests/platform/targets/tgt_dev_apis_tfm_an539/nspe/pal_crypto_config.h
@@ -39,6 +39,7 @@
 #define ARCH_TEST_RSA
 #define ARCH_TEST_RSA_1024
 #define ARCH_TEST_RSA_2048
+#define ARCH_TEST_RSA_2048_GEN_KEY
 #define ARCH_TEST_RSA_3072
 #endif
 #endif
diff --git a/api-tests/platform/targets/tgt_dev_apis_tfm_musca_a/nspe/pal_crypto_config.h b/api-tests/platform/targets/tgt_dev_apis_tfm_musca_a/nspe/pal_crypto_config.h
index c936bdd273..c04d2cf0fa 100644
--- a/api-tests/platform/targets/tgt_dev_apis_tfm_musca_a/nspe/pal_crypto_config.h
+++ b/api-tests/platform/targets/tgt_dev_apis_tfm_musca_a/nspe/pal_crypto_config.h
@@ -39,6 +39,7 @@
 #define ARCH_TEST_RSA
 #define ARCH_TEST_RSA_1024
 #define ARCH_TEST_RSA_2048
+#define ARCH_TEST_RSA_2048_GEN_KEY
 #define ARCH_TEST_RSA_3072
 #endif
 #endif
diff --git a/api-tests/platform/targets/tgt_dev_apis_tfm_musca_b1/nspe/pal_crypto_config.h b/api-tests/platform/targets/tgt_dev_apis_tfm_musca_b1/nspe/pal_crypto_config.h
index c936bdd273..c04d2cf0fa 100644
--- a/api-tests/platform/targets/tgt_dev_apis_tfm_musca_b1/nspe/pal_crypto_config.h
+++ b/api-tests/platform/targets/tgt_dev_apis_tfm_musca_b1/nspe/pal_crypto_config.h
@@ -39,6 +39,7 @@
 #define ARCH_TEST_RSA
 #define ARCH_TEST_RSA_1024
 #define ARCH_TEST_RSA_2048
+#define ARCH_TEST_RSA_2048_GEN_KEY
 #define ARCH_TEST_RSA_3072
 #endif
 #endif
diff --git a/api-tests/platform/targets/tgt_dev_apis_tfm_musca_s1/nspe/pal_crypto_config.h b/api-tests/platform/targets/tgt_dev_apis_tfm_musca_s1/nspe/pal_crypto_config.h
index 132240a68d..8a902ce0aa 100644
--- a/api-tests/platform/targets/tgt_dev_apis_tfm_musca_s1/nspe/pal_crypto_config.h
+++ b/api-tests/platform/targets/tgt_dev_apis_tfm_musca_s1/nspe/pal_crypto_config.h
@@ -39,6 +39,7 @@
 #define ARCH_TEST_RSA
 #define ARCH_TEST_RSA_1024
 #define ARCH_TEST_RSA_2048
+#define ARCH_TEST_RSA_2048_GEN_KEY
 #define ARCH_TEST_RSA_3072
 #endif
 #endif
diff --git a/api-tests/platform/targets/tgt_dev_apis_tfm_nrf/nspe/pal_crypto_config.h b/api-tests/platform/targets/tgt_dev_apis_tfm_nrf/nspe/pal_crypto_config.h
index c936bdd273..c04d2cf0fa 100644
--- a/api-tests/platform/targets/tgt_dev_apis_tfm_nrf/nspe/pal_crypto_config.h
+++ b/api-tests/platform/targets/tgt_dev_apis_tfm_nrf/nspe/pal_crypto_config.h
@@ -39,6 +39,7 @@
 #define ARCH_TEST_RSA
 #define ARCH_TEST_RSA_1024
 #define ARCH_TEST_RSA_2048
+#define ARCH_TEST_RSA_2048_GEN_KEY
 #define ARCH_TEST_RSA_3072
 #endif
 #endif
diff --git a/api-tests/platform/targets/tgt_dev_apis_tfm_psoc64/nspe/pal_crypto_config.h b/api-tests/platform/targets/tgt_dev_apis_tfm_psoc64/nspe/pal_crypto_config.h
index 132240a68d..076b65a136 100644
--- a/api-tests/platform/targets/tgt_dev_apis_tfm_psoc64/nspe/pal_crypto_config.h
+++ b/api-tests/platform/targets/tgt_dev_apis_tfm_psoc64/nspe/pal_crypto_config.h
@@ -39,6 +39,7 @@
 #define ARCH_TEST_RSA
 #define ARCH_TEST_RSA_1024
 #define ARCH_TEST_RSA_2048
+//#define ARCH_TEST_RSA_2048_GEN_KEY
 #define ARCH_TEST_RSA_3072
 #endif
 #endif
-- 
2.25.1

